Sources/Datadog/RUM/RUMTelemetry.swift

```diff
@@ -8,13 +8,22 @@ import Foundation
 
 /// Sends Telemetry events to RUM.
 ///
-/// `RUMTelemetry` complies to `Telemetry` protocol allowing
-/// sending telemetry events accross features.
+/// `RUMTelemetry` complies to `Telemetry` protocol allowing sending telemetry
+/// events accross features.
+///
+/// Events are reported up to 100 per sessions with a sampling mechanism that is
+/// configured at initialisation. Duplicates are discared.
 internal final class RUMTelemetry: Telemetry {
+    /// Maximium number of telemetry events allowed per user sessions.
+    static let MaxEventsPerSessions: Int = 100
+
     let sdkVersion: String
     let applicationID: String
     let dateProvider: DateProvider
     let dateCorrector: DateCorrectorType
+    let sampler: Sampler
+
+    private var events: [String] = []
```

Comment:
We access (rw) this array on each event and we collect events from different threads, so it needs to be synchronised to avoid crash 💥.

Also, I think we should add tests for ensuring thread safety. We have a convention for thread safety testing [using the `callConcurrently` helper](https://github.com/DataDog/dd-sdk-ios/blob/3ae90a54513dd05fc053ceb7ad84859489cae604/Tests/DatadogTests/Datadog/Core/Utils/ValuePublisherTests.swift#L133-L141), WDYT?

Reply:
Yes 👍 better to do r/w operation in the `monitor.contextProvider.async` scope.
Very cool tool this `callConcurrently` check!!